### PR TITLE
definition of the private & public server API

### PR DESCRIPTION
**Pull request by hendrikebbers**
Hi!
Thanks for adding a pull request for Dolphin Platform :)
Please add a short description that explains the changes of your pull request.
We try to review your pull request as fast as possible.

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/502)

### CODE REVIEW

**Comment by hendrikebbers** [comment]
https://github.com/canoo/dolphin-platform/issues/247

**Comment by timo0** [comment]
Reviewed 318 of 318 files at r1.
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks broke.

---

*[platform/dolphin-platform-rpm-server/src/test/java/com/canoo/impl/server/util/SimpleAnnotatedTestModel.java, line 16 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmXZ-2QSOPFHZQ-ZMtw:-KmXZ-2QSOPFHZQ-ZMtx:b-gt9gg9) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-rpm-server/src/test/java/com/canoo/impl/server/util/SimpleAnnotatedTestModel.java#L16)):*
> ```Java
>  * limitations under the License.
>  */
> package com.canoo.impl.server.util;
> ```

Did you miss some package here? com.canoo.impl looks a bit to generic to me. (Will not comment this to the other occurrences)

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/bootstrap/modules/HttpMutexModule.java, line 28 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmXZi4hXbFxpEZPCmW1:-KmXZi4hXbFxpEZPCmW2:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/bootstrap/modules/HttpMutexModule.java#L28)):*
> ```Java
>         Assert.requireNonNull(servletContext, "servletContext");
> 
>         HttpSessionMutexHolder mutexHolder = new HttpSessionMutexHolder();
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/client/ClientSessionMutextHolder.java, line 34 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmXZt-UNOW-vO6PzbN9:-KmXZt-UNOW-vO6PzbNA:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/client/ClientSessionMutextHolder.java#L34)):*
> ```Java
>     public Mutex getMutexForClientSession(final String sessionId) {
>         Assert.requireNonBlank(sessionId, "sessionId");
>         WeakReference<Mutex> mutexReference = sessionMutexMap.get(sessionId);
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/client/HttpClientSessionImpl.java, line 43 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmXZxFxnSh_27tVIvHd:-KmXZxFxnSh_27tVIvHe:b60nsfp) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/client/HttpClientSessionImpl.java#L43)):*
> ```Java
> 
>     public HttpClientSessionImpl(final String httpSessionId) {
>         this.httpSessionId = Assert.requireNonBlank(httpSessionId, "httpSessionId");
> ```

Constants?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java, line 71 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_1saehiiFxlC8MbX:-KmX_1saehiiFxlC8MbY:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java#L71)):*
> ```Java
>                 }
>             }
>             Map<String, List<String>> additionalListProperties = provider.getListProperties();
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java, line 77 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_5Xl_3ngIK9EhD7H:-KmX_5Xl_3ngIK9EhD7I:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java#L77)):*
> ```Java
>                 }
>             }
>             Map<String, Boolean> additionalBooleanProperties = provider.getBooleanProperties();
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java, line 83 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_6ETqcC3d5HGy7h-:-KmX_6ETqcC3d5HGy7h0:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java#L83)):*
> ```Java
>                 }
>             }
>             Map<String, Integer> additionalIntegerProperties = provider.getIntegerProperties();
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java, line 89 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_6rG8pnP3flMh0Fa:-KmX_6rG8pnP3flMh0Fb:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java#L89)):*
> ```Java
>                 }
>             }
>             Map<String, Long> additionalLongProperties = provider.getLongProperties();
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultModuleConfig.java, line 60 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_K0UinF8HBb7E2fT:-KmX_K0UinF8HBb7E2fU:b60nsfp) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultModuleConfig.java#L60)):*
> ```Java
> 
>     public static int getSessionTimeout(final PlatformConfiguration configuration) {
>         return Assert.requireNonNull(configuration, "configuration").getIntProperty(SESSION_TIMEOUT, SESSION_TIMEOUT_DEFAULT_VALUE);
> ```

Constants?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java, line 103 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_Ty4LcFkNrDXKrVT:-KmX_Ty4LcFkNrDXKrVU:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java#L103)):*
> ```Java
>             setProperty(key, "");
>         } else {
>             StringBuilder builder = new StringBuilder();
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java, line 121 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_WLh0f3P_Szk7gsY:-KmX_WLik-RXsaPApgTv:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java#L121)):*
> ```Java
> 
>     public void log() {
>         Set<Map.Entry<Object, Object>> properties = internalProperties.entrySet();
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java, line 21 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_dONrtR_hoEE-Afr:-KmX_dONrtR_hoEE-Afs:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java#L21)):*
> ```Java
>     public void sessionCreated(HttpSessionEvent sessionEvent) {
>         Assert.requireNonNull(sessionEvent, "sessionEvent");
>         HttpSession session = sessionEvent.getSession();
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java, line 31 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_i-bCEVqn-Z0zrkR:-KmX_i-cfkn9dODoly1T:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java#L31)):*
> ```Java
>     public void sessionDestroyed(HttpSessionEvent sessionEvent) {
>         Assert.requireNonNull(sessionEvent, "sessionEvent");
>         HttpSession session = sessionEvent.getSession();
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java, line 38 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_jeXo8x3CYeDRLFI:-KmX_jeXo8x3CYeDRLFJ:b-qsyuya) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java#L38)):*
> ```Java
>     public Mutex getMutexForHttpSession(final String sessionId) {
>         Assert.requireNonBlank(sessionId, "sessionId");
>         WeakReference<Mutex> mutexReference = sessionMutexMap.get(sessionId);
> ```

final?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/platform/server/spi/AbstractBaseModule.java, line 17 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmX_t_izTX3HsRd1-Ai:-KmX_t_izTX3HsRd1-Aj:b-w9bwwv) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/platform/server/spi/AbstractBaseModule.java#L17)):*
> ```Java
>     @Override
>     public boolean shouldBoot(PlatformConfiguration configuration) {
>         return Assert.requireNonNull(configuration, "configuration").getBooleanProperty(getActivePropertyName(), true);
> ```

Constant from above?

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/platform/server/spi/ConfigurationProviderAdapter.java, line 7 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmXa-j9hHf7J7HiAHDk:-KmXa-j9hHf7J7HiAHDl:b2ngvnt) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/platform/server/spi/ConfigurationProviderAdapter.java#L7)):*
> ```Java
> import java.util.Map;
> 
> /**
> ```

...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/canoo/dolphin-platform/502)*

**Comment by timo0** [comment]
Reviewed 318 of 318 files at r1.
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/canoo/dolphin-platform/502)*

**Comment by timo0** [comment]
Reviewed 318 of 318 files at r1.
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/canoo/dolphin-platform/502)*

File: platform-extras/dolphin-platform-distributed-eventbus/src/main/java/com/canoo/impl/server/event/DefaultHazelcastProvider.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.canoo.dolphin.server.event;
+package com.canoo.impl.server.event;
```
Comment (timo0) [inline platform-extras/dolphin-platform-distributed-eventbus/src/main/java/com/canoo/impl/server/event/DefaultHazelcastProvider.java]: Did you miss some package here? com.canoo.impl looks a bit to generic to me. (Will not comment this to the other occurrences)

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/bootstrap/modules/HttpMutexModule.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        return HTTP_MUTEX_MODULE_ACTIVE;
+    }
+
+    @Override
+    public void initialize(final ServerCoreComponents coreComponents) throws ModuleInitializationException {
+        Assert.requireNonNull(coreComponents, "coreComponents");
+        final ServletContext servletContext = coreComponents.getServletContext();
+        Assert.requireNonNull(servletContext, "servletContext");
+
+        HttpSessionMutexHolder mutexHolder = new HttpSessionMutexHolder();
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/bootstrap/modules/HttpMutexModule.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/client/ClientSessionMutextHolder.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Override
+    public void sessionDestroyed(ClientSession clientSession) {
+        Assert.requireNonNull(clientSession, "clientSession");
+        sessionMutexMap.remove(clientSession.getId());
+    }
+
+    public Mutex getMutexForClientSession(final String sessionId) {
+        Assert.requireNonBlank(sessionId, "sessionId");
+        WeakReference<Mutex> mutexReference = sessionMutexMap.get(sessionId);
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/client/ClientSessionMutextHolder.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java [hunk 4]
```diff
@@ -50,22 +52,46 @@ private ConfigurationFileLoader() {
...
         for(ConfigurationProvider provider : serviceLoader) {
-            Map<String, String> additionalProperties = provider.getProperties();
-            for(Map.Entry<String, String> property : additionalProperties.entrySet()) {
+            Map<String, String> additionalStringProperties = provider.getStringProperties();
+            for(Map.Entry<String, String> property : additionalStringProperties.entrySet()) {
                 if(!configuration.containsProperty(property.getKey())) {
                     configuration.setProperty(property.getKey(), property.getValue());
                 }
             }
+            Map<String, List<String>> additionalListProperties = provider.getListProperties();
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    public void sessionDestroyed(HttpSessionEvent sessionEvent) {
+        Assert.requireNonNull(sessionEvent, "sessionEvent");
+        HttpSession session = sessionEvent.getSession();
+        Assert.requireNonNull(session, "session");
+        sessionMutexMap.remove(session.getId());
+    }
+
+    public Mutex getMutexForHttpSession(final String sessionId) {
+        Assert.requireNonBlank(sessionId, "sessionId");
+        WeakReference<Mutex> mutexReference = sessionMutexMap.get(sessionId);
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        Assert.requireNonNull(session, "session");
+        final Mutex mutex = new Mutex();
+        session.setAttribute(SESSION_MUTEX_ATTRIBUTE, mutex);
+        sessionMutexMap.put(session.getId(), new WeakReference<>(mutex));
+    }
+
+    @Override
+    public void sessionDestroyed(HttpSessionEvent sessionEvent) {
+        Assert.requireNonNull(sessionEvent, "sessionEvent");
+        HttpSession session = sessionEvent.getSession();
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+public class HttpSessionMutexHolder implements HttpSessionListener {
+
+    private final Map<String, WeakReference<Mutex>> sessionMutexMap = new HashMap<>();
+
+    private final static String SESSION_MUTEX_ATTRIBUTE = "Session-Mutex";
+
+    @Override
+    public void sessionCreated(HttpSessionEvent sessionEvent) {
+        Assert.requireNonNull(sessionEvent, "sessionEvent");
+        HttpSession session = sessionEvent.getSession();
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/platform/server/spi/ConfigurationProviderAdapter.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
+package com.canoo.platform.server.spi;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/platform/server/spi/ConfigurationProviderAdapter.java]: ...

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java [hunk 9]
```diff
@@ -50,22 +52,46 @@ private ConfigurationFileLoader() {
...
+            for(Map.Entry<String, List<String>> property : additionalListProperties.entrySet()) {
+                if(!configuration.containsProperty(property.getKey())) {
+                    configuration.setListProperty(property.getKey(), property.getValue());
+                }
+            }
+            Map<String, Boolean> additionalBooleanProperties = provider.getBooleanProperties();
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java [hunk 10]
```diff
@@ -50,22 +52,46 @@ private ConfigurationFileLoader() {
...
+            for(Map.Entry<String, Boolean> property : additionalBooleanProperties.entrySet()) {
+                if(!configuration.containsProperty(property.getKey())) {
+                    configuration.setBooleanProperty(property.getKey(), property.getValue());
+                }
+            }
+            Map<String, Integer> additionalIntegerProperties = provider.getIntegerProperties();
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java [hunk 11]
```diff
@@ -50,22 +52,46 @@ private ConfigurationFileLoader() {
...
+            for(Map.Entry<String, Integer> property : additionalIntegerProperties.entrySet()) {
+                if(!configuration.containsProperty(property.getKey())) {
+                    configuration.setIntProperty(property.getKey(), property.getValue());
+                }
+            }
+            Map<String, Long> additionalLongProperties = provider.getLongProperties();
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+    public void setProperty(final String key, final String value) {
+        if (value == null) {
+            LOG.warn("Setting property '{}' to null value will be ignored.");
+        } else {
+            internalProperties.setProperty(key, value);
+        }
+    }
+
+    public void log() {
+        Set<Map.Entry<Object, Object>> properties = internalProperties.entrySet();
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java]: final?

File: platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+        setProperty(key, Boolean.toString(value));
+    }
+
+    public void setListProperty(final String key, final List<String> values) {
+        if (values == null) {
+            setProperty(key, null);
+        } else if (values.isEmpty()) {
+            setProperty(key, "");
+        } else {
+            StringBuilder builder = new StringBuilder();
```
Comment (timo0) [inline platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java]: final?

**Comment by hendrikebbers** [comment]
Review status: 0 of 190 files reviewed at latest revision, 13 unresolved discussions, some commit checks broke.

---

*[platform-extras/dolphin-platform-distributed-eventbus/src/main/java/com/canoo/impl/server/event/DefaultHazelcastProvider.java, line 16 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYwdLlAcw7EqmD_-r1-16:-Km_TVkMiRcipL5NKLlj:b-j3yo4n) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform-extras/dolphin-platform-distributed-eventbus/src/main/java/com/canoo/impl/server/event/DefaultHazelcastProvider.java#L16)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

Did you miss some package here? com.canoo.impl looks a bit to generic to me. (Will not comment this to the other occurrences)
</blockquote></details>

Yeah, I know. Currently it's like this:

"com.canoo.platform.*" contains the public API
"com.canoo.impl.*" contains the private API

I think the most important point is the package of the public API, since we can change the package of the private API any time ;)

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/bootstrap/modules/HttpMutexModule.java, line 28 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLUr-r1-28:-Km_TsUNFvPE6r3eo1OV:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/bootstrap/modules/HttpMutexModule.java#L28)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/client/ClientSessionMutextHolder.java, line 34 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLUu-r1-34:-Km_TzjgscNwCqJCPFdA:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/client/ClientSessionMutextHolder.java#L34)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java, line 71 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLUw-r1-71:-Km_UKfq7fwm8-wFHUXN:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java#L71)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java, line 77 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLUw-r1-77:-Km_ULTwBpg5oSpP3qAR:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java#L77)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java, line 83 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLUw-r1-83:-Km_UMQrSPYedqigZZtT:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java#L83)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java, line 89 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLUw-r1-89:-Km_UNN13ObQ6MpSkGkA:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/ConfigurationFileLoader.java#L89)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java, line 103 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLUy-r1-103:-Km_UiN2k5fYixGKGUnV:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java#L103)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java, line 121 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLUy-r1-121:-Km_Uj8NwtTASGne55aY:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/config/DefaultPlatformConfiguration.java#L121)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java, line 21 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLV0-r1-21:-Km_UqYdQqCXEpiRBnTX:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java#L21)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java, line 31 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLV0-r1-31:-Km_UrVX_i00BOwFZ1qQ:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java#L31)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java, line 38 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLV0-r1-38:-Km_Us4fbZ_yd3wGlRqm:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/impl/server/servlet/HttpSessionMutexHolder.java#L38)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

final?
</blockquote></details>

done

---

*[platform/dolphin-platform-server/src/main/java/com/canoo/platform/server/spi/ConfigurationProviderAdapter.java, line 7 at r1](https://reviewable.io:443/reviews/canoo/dolphin-platform/502#-KmRrVYz4xNymWD8SLV7-r1-7:-Km_UwtEFQ40GHKya4t6:bb7f2mf) ([raw file](https://github.com/canoo/dolphin-platform/blob/97a8da301ffc9fd73c4d709d87197905dc78d19a/platform/dolphin-platform-server/src/main/java/com/canoo/platform/server/spi/ConfigurationProviderAdapter.java#L7)):*
<details><summary><i>Previously, timo0 (Timo Brandstätter) wrote…</i></summary><blockquote>

...
</blockquote></details>

done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/canoo/dolphin-platform/502)*

**Pull request merged**